.ci/workchains.py

```diff
@@ -7,10 +7,62 @@
 # For further information on the license, see the LICENSE.txt file        #
 # For further information please visit http://www.aiida.net               #
 ###########################################################################
-
-from aiida.engine import calcfunction, workfunction, WorkChain, ToContext, append_
+from aiida.common import AttributeDict
+from aiida.engine import calcfunction, workfunction, WorkChain, ToContext, append_, while_
+from aiida.engine import BaseRestartWorkChain, register_process_handler, ProcessHandlerReport
 from aiida.engine.persistence import ObjectLoader
 from aiida.orm import Int, List, Str
+from aiida.plugins import CalculationFactory
+
+
+ArithmeticAddCalculation = CalculationFactory('arithmetic.add')
+
+
+class ArithmeticAddBaseWorkChain(BaseRestartWorkChain):
+    """Ridiculous workchain around `AritmethicAddCalculation` with automated sanity checks and error handling."""
+
+    _process_class = ArithmeticAddCalculation
+
+    @classmethod
+    def define(cls, spec):
+        """Define the process specification."""
+        super(ArithmeticAddBaseWorkChain, cls).define(spec)
+        spec.expose_inputs(ArithmeticAddCalculation, namespace='add')
+        spec.expose_outputs(ArithmeticAddCalculation)
+        spec.outline(
+            cls.setup,
+            while_(cls.should_run_process)(
+                cls.run_process,
+                cls.inspect_process,
+            ),
+            cls.results,
+        )
+        spec.exit_code(100, 'ERROR_TOO_BIG', message='The sum was too big.')
+
+    def setup(self):
+        """Call the `setup` of the `BaseRestartWorkChain` and then create the inputs dictionary in `self.ctx.inputs`.
+
+        This `self.ctx.inputs` dictionary will be used by the `BaseRestartWorkChain` to submit the process in the
+        internal loop.
+        """
+        super(ArithmeticAddBaseWorkChain, self).setup()
+        self.ctx.inputs = AttributeDict(self.exposed_inputs(ArithmeticAddCalculation, 'add'))
+
+
+@register_process_handler(ArithmeticAddBaseWorkChain, priority=500)
+def sanity_check_not_too_big(self, node):
+    """My puny brain cannot deal with numbers that I cannot count on my hand."""
+    if node.is_finished_ok and node.outputs.sum > 10:
+        return ProcessHandlerReport(True, self.exit_codes.ERROR_TOO_BIG)
+
+
+@register_process_handler(ArithmeticAddBaseWorkChain, priority=400)
+def error_negative_sum(self, node):
+    """What even is a negative number, how can I have minus three melons?!."""
+    if node.exit_status == ArithmeticAddCalculation.exit_codes.ERROR_NEGATIVE_NUMBER.status:
```

Comment:
I think this error isn't defined.

Reply:
I added it in the [first commit](e22ad2bdafb8d12eaa4431381f47dc800214efa3) of this PR